front/src/applications/stdcm/hooks/useStdcmConsist.ts

```diff
@@ -15,28 +24,55 @@ const useStdcmConsist = () => {
   const totalMass = useSelector(getTotalMass);
   const onTotalMassChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalMassValue = Number(e.target.value);
+    setTotalMassChanged(true);
     dispatch(updateTotalMass(totalMassValue === 0 ? undefined : totalMassValue));
   };
 
   const totalLength = useSelector(getTotalLength);
   const onTotalLengthChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalLengthValue = Number(e.target.value);
+    setTotalLengthChanged(true);
     dispatch(updateTotalLength(totalLengthValue === 0 ? undefined : totalLengthValue));
   };
 
   const maxSpeed = useSelector(getMaxSpeed);
   const onMaxSpeedChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const totalMaxSpeed = Number(e.target.value);
+    setMaxSpeedChanged(true);
     dispatch(updateMaxSpeed(totalMaxSpeed === 0 ? undefined : totalMaxSpeed));
   };
 
+  const prefillConsist = (
+    rollingStock?: LightRollingStockWithLiveries,
+    towed?: TowedRollingStock
+  ) => {
+    if (!rollingStock && !towed) {
+      return;
+    }
```

Comment:
If I clear the tower input and then I clear the rolling stock input, we pass here whereas we should clear all (unaffected by the user) inputs